### DISCUSSION TITLE
Would it be possible virtualize iOS?

### DISCUSSION THREAD

**Discussion by raspu** [utmapp/UTM#3244] [UTM Help (macOS)]
Would it be theoretically possible to use UTM to virtualize iOS (either on macOS or on iOS)? I know [Corellium](https://www.corellium.com/) was able to achieve something similar using ARM-virtual devices, and I am curious if the same could be achieved with actual devices or with an M1 mac.

**Comment by osy** [comment]
Not until iOS support in QEMU gets beyond booting the kernel...

**Reply by raspu** [reply]
So the current restriction is QEMU support? Interesting that it can actually boot the kernel, tho...

**Reply by raspu** [reply]
Thanks for replying!
